src/test/java/uk/nhs/cdss/resourceProviders/CarePlanProviderComponentTest.java

```diff
@@ -0,0 +1,87 @@
+package uk.nhs.cdss.resourceProviders;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.mockito.Mockito.when;
+
+import ca.uhn.fhir.parser.IParser;
+import ca.uhn.fhir.rest.param.ReferenceParam;
+import ca.uhn.fhir.rest.server.exceptions.InvalidRequestException;
+import java.util.Collection;
+import java.util.Collections;
+import org.hamcrest.CustomTypeSafeMatcher;
+import org.hamcrest.Matcher;
+import org.hl7.fhir.dstu3.model.CarePlan;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.nhs.cdss.entities.ResourceEntity;
+import uk.nhs.cdss.entities.ResourceEntity.IdVersion;
+import uk.nhs.cdss.fixtures.CarePlanFixtures;
+import uk.nhs.cdss.repos.ResourceRepository;
+
+@SpringBootTest
+@RunWith(SpringRunner.class)
+@TestPropertySource(locations = "classpath:application-test.properties")
+public class CarePlanProviderComponentTest {
+
+  @Autowired
+  private CarePlanProvider carePlanProvider;
+
+  @Autowired
+  private IParser fhirParser;
+
+  @MockBean
+  private ResourceRepository resourceRepository;
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Test
+  public void findsCarePlan() {
+    CarePlan expected = CarePlanFixtures.carePlan();
+    ResourceEntity resourceEntity = ResourceEntity.builder()
+        .resourceType(ResourceType.CarePlan)
+        .idVersion(new IdVersion(1L , 1L))
+        .resourceJson(fhirParser.encodeResourceToString(expected))
+        .build();
+    when(resourceRepository.findAll()).thenReturn(Collections.singletonList(resourceEntity));
+
+    Collection<CarePlan> results = carePlanProvider
+        .findByEncounterContext(referenceParam(expected));
+
+    assertThat(results, contains(carePlan(expected)));
+  }
+
+  @Test
+  public void failsInvalidSearch() {
+    expectedException.expect(InvalidRequestException.class);
```

Comment:
If you're not checking any details of the exception, is there a reason to use ExpectedException rather than the `@Test(expected=` annotation? Either is fine, I just find it easier to read the annotation due to past habits.

Reply:
I tend just to use the expected one as it gives you more control over when the exception is expected. In this case I guess it makes no difference.